packages/flutter_driver/lib/src/driver.dart

```diff
@@ -384,6 +385,13 @@ class FlutterDriver {
     return GetTextResult.fromJson(await _sendCommand(new GetText(finder, timeout: timeout))).text;
   }
 
+  /// Sends a string and returns a string.
+  ///
+  /// The application can respond to this by providing a handler to [enableFlutterDriverExtension].
+  Future<String> requestData(String message, { Duration timeout }) async {
```

Comment:
I've changed DataHandler to return a future string.

Any ideas what requestData, RequestData, DataHandler, and RequestDataResult should be called, if not those? I couldn't come up with anything more generic...